controller/authController.js

```diff
@@ -0,0 +1,197 @@
+const { authModel } = require('../models/auth');
+const { otpModel } = require('../models/otp');
+const { StatusCodes } = require('http-status-codes');
+const { RESPONSE_STATUS } = require('../utils/enum');
+const { GeneralResponse } = require('../utils/response');
+const { Messages } = require('../utils/messages');
+const { GeneralError } = require('../utils/error');
+const { generateToken } = require('../middleware/authentication');
+const { sendOtp } = require('../middleware/mail');
+const bcrypt = require('bcrypt');
+const moment = require('moment');
+const logger = require('../logger/logger');
+
+const registration = async (req, res, next) => {
+  const encrypt = await bcrypt.hash(req.body.password, 10);
+  const { name, email, phone_number, status, role } = req.body;
+  const existingUser = await authModel.findOne({ email: req.body.email });
+  if (existingUser) {
+    next(
+      new GeneralError(
+        `User ${Messages.ALREADY_EXIST}`,
+        undefined,
+        StatusCodes.CONFLICT,
+        RESPONSE_STATUS.ERROR,
+      ),
+    );
+  }
+  if (!req.file) {
+    next(
+      new GeneralError(
+        Messages.IMAGE_NOT_FOUND,
+        StatusCodes.NOT_FOUND,
+        undefined,
+        RESPONSE_STATUS.ERROR,
+      ),
+    );
+  }
+  const newUser = new authModel({
+    name,
+    email,
+    password: encrypt,
+    phone_number,
+    profile_image: req.file.filename,
+    status,
+    role,
+  });
+  const user = await newUser.save();
+  if (user) {
+    next(
+      new GeneralResponse(
```

Comment:
GeneralResponse function parameter is different sequence so please make sure set parameter as main function parameter sequence